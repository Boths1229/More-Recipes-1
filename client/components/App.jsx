/* eslint-disable react/prefer-stateless-function */
import React, { PropTypes } from 'react';


const propTypes = {
  children: PropTypes.shape().isRequired
};

/**
 * @description A class to mount all component
 *
 * @extends Component
 */
class App extends React.Component {
  /**
   * @memberof App
   *
   * @returns {void} returns the component to be mounted
  */
  render() {
    return (
      <div>
        {this.props.children}
      </div>
    );
  }
}

App.propTypes = propTypes;

export default App;
