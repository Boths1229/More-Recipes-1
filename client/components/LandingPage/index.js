import React from 'react';
import Header from './header';
import Carousel from './carousel';
import AboutSection from './aboutSection';
import Footer from '../common/footer';

/* eslint-disable no-unused-vars */
/* eslint-disable require-jsdoc */
class LandingPage extends React.Component {
  render() {
    return (
      <div>
        <Header/>
        <Carousel/>
        <AboutSection/>
        <Footer/>
      </div>
      
    );
  }
}

export default LandingPage;