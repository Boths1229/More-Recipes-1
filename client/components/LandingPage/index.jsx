import React from 'react';
import Header from './Header';
import Carousel from './Carousel';
import AboutSection from './AboutSection';
import Footer from '../common/Footer';


/* eslint-disable react/prefer-stateless-function */

/**
 * @description A class to mount all components related to LandingPage
 *
 * @extends Component
 */
class LandingPage extends React.Component {
  /**
   * @description renders the components
   *
   * @memberof LandingPage
   *
   * @returns {JSX} returns the components
  */
  render() {
    return (
      <div>
        <Header />
        <Carousel />
        <AboutSection />
        <Footer />
      </div>
    );
  }
}

export default LandingPage;
