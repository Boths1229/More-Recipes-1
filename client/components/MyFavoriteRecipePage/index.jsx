import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Spinner } from 'react-preloading-component';
import Header from '../common/AuthHeader';
import Footer from '../common/Footer';
import FavRecipeCard from './FavoriteRecipeCard';
import { getUserFavRecipes } from '../../actionsCreator/recipes';
import verifyUser from '../../util/Authentication';


const propTypes = {
  actions: PropTypes.shape({
    getUserFavRecipes: PropTypes.func
  }).isRequired,
  user: PropTypes.shape({
    id: PropTypes.number,
    username: PropTypes.string
  }).isRequired,
  favoriteRecipes: PropTypes.oneOfType([
    PropTypes.shape(),
    PropTypes.arrayOf(PropTypes.shape())
  ]).isRequired,
  isLoading: PropTypes.bool
};

const defaultProps = {
  isLoading: null
};

/**
 * @description A class to mount all components related to MyFavoriteRecipePage
 * @extends Component
 */
export class MyFavoriteRecipePage extends Component {
  /**
   * handles rendering favorite recipe
   * @param {object} props
  */
  constructor(props) {
    super(props);

    this.renderFavoriteRecipes = this.renderFavoriteRecipes.bind(this);
  }

  /**
   * @description performs an action right after the component mount
   *
   * @memberof MyFavoriteRecipePage
   *
   * @returns {void}
  */
  componentDidMount() {
    if (verifyUser() === true) {
      this.props.actions.getUserFavRecipes(this.props.user.id);
    }
  }

  /**
   * @memberof MyFavoriteRecipePage
   *
   * @returns {void}
  */
  renderFavoriteRecipes() {
    const {
      isLoading,
      favoriteRecipes,
      user
    } = this.props;
    if (isLoading) {
      return (
        <div style={{
          marginTop: '150px',
          marginLeft: 'auto',
          marginRight: 'auto',
        }}
        >
          <Spinner />
        </div>
      );
    }
    if (favoriteRecipes.message || favoriteRecipes.favorited.length === 0) {
      return (
        <h1 style={{ margin: '150px auto 0', width: '100%', textAlign: 'center' }}>
          You have no favorite recipes
        </h1>
      );
    }
    return favoriteRecipes.favorited.map(favorites => (
      <FavRecipeCard
        key={favorites.id}
        recipe={favorites.Recipe}
        favorite={this.favoriteRecipe}
        userId={user.id}
      />
    ));
  }

  /**
   * @description renders the components
   *
   * @memberof MyFavoriteRecipePage
   *
   * @returns {JSX} returns the components
  */
  render() {
    return (
      <div>
        <Header user={this.props.user} />
        <div className="container">
          <div className="row section-popular">
            <div className="col-sm-12">
              <h2 className="pt-3 headline centered">My Favorite Recipes</h2>
            </div>
          </div>
          <div className="row pt-4">
            {
              this.renderFavoriteRecipes()
            }
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

/**
 * @description maps state to properties of MyFavoriteRecipePage
 *
 * @param  {object} state
 *
 * @returns {object} returns the state to be mapped to props
 */
function mapStateToProps(state) {
  return {
    user: state.auth.user,
    favoriteRecipes: state.recipes.favoriteRecipes,
    isLoading: state.recipes.isLoading
  };
}

/**
 * @description maps action to properties of MyFavoriteRecipePage
 *
 * @param  {object} dispatch
 *
 * @returns {object} returns the action to be bind
 */
function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      getUserFavRecipes
    }, dispatch)
  };
}

MyFavoriteRecipePage.propTypes = propTypes;
MyFavoriteRecipePage.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(MyFavoriteRecipePage);
