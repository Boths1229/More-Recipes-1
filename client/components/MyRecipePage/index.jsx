/* eslint-disable no-unused-vars */
/**
 *  @fileOverview renders the user recipes page
 *
 *  @author       Sanni Taiwo
 *
 *  @requires     NPM:react
 *  @requires     NPM:react-redux
 *  @requires     NPM:redux
 *  @requires     NPM:toastr
 */
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import toastr from 'toastr';
import Header from '../common/authHeader';
import Footer from '../common/footer';
import MyRecipeCard from './MyRecipeCard';
import EditRecipeModal from './EditRecipeModal';
import DeleteRecipeModal from './deleteRecipeModal';
import { logoutAction } from '../../actions/authAction';
import verifyUser from '../../util/Authentication';
import {
  getUserRecipes,
  editRecipeAction,
  deleteRecipeAction,
} from '../../actions/recipesAction';

const propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number,
    username: PropTypes.string
  }),
  actions: PropTypes.shape({
    logoutAction: PropTypes.func,
    getUserRecipes: PropTypes.func,
    deleteRecipeAction: PropTypes.func,
    editRecipeAction: PropTypes.func
  }).isRequired,
  userRecipes: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};
const defaultProps = {
  user: null
};


/**
 * @description A class to mount all components related to MyRecipePage
 * @extends Component
 */
class MyRecipePage extends Component {
  /**
   * handles editing recipe
   * @param {object} props
   * @param {object} context
  */
  constructor(props, context) {
    super(props, context);
    this.state = {
      recipe: {
        recipeName: '',
      },
      index: '',
      errorMessage: ''
    };

    this.renderUserRecipes = this.renderUserRecipes.bind(this);
    this.setRecipe = this.setRecipe.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onFocus = this.onFocus.bind(this);
    this.handleDeletion = this.handleDeletion.bind(this);
  }

  /**
   * @description performs an action right after the component mount
   *
   * @memberof MyRecipePage
   *
   * @returns {void}
  */
  componentDidMount() {
    if (verifyUser() === true) {
      this.props.actions.getUserRecipes(this.props.user.id);
    }
  }

  /**
   * @description removes errorMessage when focused
   *
   * @param {object} event
   *
   * @memberof MyRecipePage
   *
   * @returns {void}
  */
  onFocus(event) {
    this.setState({ errorMessage: '' });
  }

  /**
   * @description get the current recipe from the user recipes
   *
   * @param {number} index
   *
   * @memberof MyRecipePage
   *
   * @returns {void}
  */
  setRecipe(index) {
    this.setState({
      recipe: this.props.userRecipes[index],
      index,
    });
  }

  /**
   * @description handles the deletion of recipe
   *
   * @param {object} event
   *
   * @memberof MyRecipePage
   *
   * @returns {void}
  */
  handleDeletion(event) {
    const { recipe, index } = this.state;
    this.props.actions.deleteRecipeAction(recipe.id, index)
      .then(() => {
        toastr.success('Recipe successfully deleted');
        $('#deleteModal').modal('hide');
      });
  }

  /**
   * @description handles the submission of form
   *
   * @param {object} event
   *
   * @memberof MyRecipePage
   *
   * @returns {void}
  */
  handleSubmit(event) {
    const { recipe, index } = this.state;
    event.preventDefault();
    this.props.actions.editRecipeAction(recipe.id, recipe, index)
      .then(() => {
        toastr.success('Recipe successfully updated');
        $('#editModal').modal('hide');
      })
      .catch((err) => {
        const { data } = err.response;
        /* eslint-disable array-callback-return */
        data.map((message) => {
          this.setState({
            errorMessage: message
          });
        });
      });
  }


  /**
   * @description checks for update in form entry
   *
   * @param {object} event
   *
   * @memberof MyRecipePage
   *
   * @returns {void}
  */
  handleChange(event) {
    const { name, value } = event.target;
    this.setState({
      recipe: {
        ...this.state.recipe,
        [name]: value
      }
    });
  }


  /**
   * @description checks if a user have recipes
   *
   * @memberof MyRecipePage
   *
   * @returns {void} returns a components
  */
  renderUserRecipes() {
    if (this.props.userRecipes.length === 0) {
      return (
        <h1>
          You have no Recipe
        </h1>
      );
    }
    return this.props.userRecipes.map((recipes, i) => (
      <MyRecipeCard
        key={recipes.id}
        index={i}
        recipeDetails={recipes}
        getRecipe={this.setRecipe}
      />
    ));
  }

  /**
   * @description renders the components
   *
   * @memberof MyRecipePage
   *
   * @returns {void} returns the components
  */
  render() {
    return (
      <div>
        <Header user={this.props.user} />
        <div className="container">
          <div className="row section-popular">
            <div className="col-sm-12">
              <h2 className="pt-3 headline centered">My Recipes</h2>
            </div>
          </div>
          <div className="row pt-4">
            { this.renderUserRecipes()}
          </div>
          <EditRecipeModal
            details={this.state.recipe}
            handleChange={this.handleChange}
            handleSubmit={this.handleSubmit}
            errorMessage={this.state.errorMessage}
            onFocus={this.onFocus}
          />
          <DeleteRecipeModal
            handleDeletion={this.handleDeletion}
          />
        </div>
        <Footer />
      </div>
    );
  }
}

MyRecipePage.propTypes = propTypes;

MyRecipePage.defaultProps = defaultProps;

MyRecipePage.contextTypes = {
  router: PropTypes.object
};

/**
 * @description maps state to properties of MyRecipePage
 *
 * @param  {object} state
 *
 * @returns {object} returns the state to be mapped to props
 */
function mapStateToProps(state) {
  return {
    user: state.auth.user,
    userRecipes: state.recipes.userRecipes
  };
}

/**
 * @description maps action to properties of MyRecipePage
 *
 * @param  {object} dispatch
 *
 * @returns {object} returns the action to be bind
 */
function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      getUserRecipes,
      editRecipeAction,
      deleteRecipeAction,
      logoutAction
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(MyRecipePage);
