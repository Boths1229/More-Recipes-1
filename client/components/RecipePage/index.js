import React from 'react';
import Header from '../common/header';
import RecipeCardGrid from './recipeCardGrid';
import PopularRecipeCardList from './popularRecipeCardList';



/* eslint-disable no-unused-vars */
/* eslint-disable require-jsdoc */
class RecipePage extends React.Component {

  render() {
    return (
      <div>
        <Header/>
        <div className="content">
          <div className="container cont_area">
            <div className="row">
              <RecipeCardGrid/>
              <PopularRecipeCardList/>
            </div>
          </div>
        </div>
      </div>         
    );
  }
}

export default RecipePage;