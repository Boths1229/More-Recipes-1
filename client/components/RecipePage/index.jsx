/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Header from '../common/header';
import Footer from '../common/footer';
import RecipeCardGrid from './recipeCardGrid';
import PopularRecipeCardList from './popularRecipeCardList';
import { getAllRecipesAction,
  getPopularRecipesAction,
} from '../../actions/recipesAction';

/* eslint-disable require-jsdoc */
/* eslint-disable class-methods-use-this */
class RecipePage extends Component {
  componentWillMount() {
    this.props.actions.getAllRecipesAction();
    this.props.actions.getPopularRecipesAction();
  }

  render() {
    const {
      popularRecipes,
    } = this.props;
    return (
      <div>
        <Header/>
        <div className="content">
          <div className="container cont_area">
            <div className="row">
              <RecipeCardGrid
              allRecipes={this.props.recipes}
              upvoteAction={this.upvoteRecipe}
              />
              <PopularRecipeCardList popularRecipes={popularRecipes}/>
            </div>
          </div>
        </div>
        <Footer/>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    recipes: state.recipes.recipes,
    popularRecipes: state.recipes.popularRecipes,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      getAllRecipesAction,
      getPopularRecipesAction,
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(RecipePage);
