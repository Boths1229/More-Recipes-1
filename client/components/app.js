import React from 'react';

/* eslint-disable no-unused-vars */
/* eslint-disable require-jsdoc */
class App extends React.Component {
  render() {
    return (
      <div>
        {this.props.children}
      </div>
    );
  }
}

export default App;
