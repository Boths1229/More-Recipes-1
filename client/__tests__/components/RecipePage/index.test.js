import React from 'react';
import { shallow } from 'enzyme';
import { RecipePage } from '../../../components/RecipePage';
import MockData from '../../__mocks__/actions/recipes';

let props;

localStorage.clear = () => {};
localStorage.getItem = () => {};
localStorage.setItem = () => {};

const setup = () => {
  props = {
    popularRecipes: MockData.getPopularRecipesSuccess,
    recipes: MockData.getAllRecipesSuccess.allRecipes,
    page: MockData.getAllRecipesSuccess.pages,
    user: MockData.userData.profile,
    actions: {
      getAllRecipesAction: () => {},
      getPopularRecipesAction: () => {},
      changeAuthAction: () => {},
    },
    isauthenticated: true,
    isLoading: false,

  };

  return shallow(<RecipePage {...props} />);
};


describe('RecipePage component', () => {
  it('should render correctly for logged in user', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly for user not logged in', () => {
    const wrapper = setup();
    wrapper.setProps({ isauthenticated: false });
    expect(wrapper).toMatchSnapshot();
  });
});
