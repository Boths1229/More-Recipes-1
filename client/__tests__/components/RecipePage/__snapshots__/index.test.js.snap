// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipePage component should render correctly for logged in user 1`] = `
<div>
  <Header />
  <div
    className="content"
  >
    <div
      className="container cont_area"
    >
      <div
        className="row"
      >
        <RecipeCardGrid
          allRecipes={
            Array [
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-02-20T10:10:45.197Z",
                "description": "amazing recipe",
                "downvotes": 1,
                "id": 26,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F702446f3-a2c2-48c7-a9d1-af20be9281d7.jpeg?alt=media&token=1827adb2-4108-4939-9fe0-778e27dd5212",
                "ingredients": "rice,chicken,rice,chicken",
                "mealType": "lunch",
                "method": "research on that",
                "recipeName": "rice and chicken",
                "updatedAt": "2018-03-12T09:32:31.056Z",
                "upvotes": 2,
                "userId": 1,
              },
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-02-20T10:11:25.020Z",
                "description": "amazing",
                "downvotes": 1,
                "id": 27,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F702446f3-a2c2-48c7-a9d1-af20be9281d7.jpeg?alt=media&token=1827adb2-4108-4939-9fe0-778e27dd5212",
                "ingredients": "awesome",
                "mealType": "breakfast",
                "method": "great",
                "recipeName": "new recipe",
                "updatedAt": "2018-03-06T11:08:34.739Z",
                "upvotes": 0,
                "userId": 1,
              },
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-02-20T10:19:50.363Z",
                "description": "test",
                "downvotes": 0,
                "id": 33,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F463fb7f5-37ad-4a76-8dce-2ad3feb73e00.jpeg?alt=media&token=6b3afedd-09f3-4582-829a-6a665454db8d",
                "ingredients": "great",
                "mealType": "tea",
                "method": "great",
                "recipeName": "recipe",
                "updatedAt": "2018-03-06T10:57:55.712Z",
                "upvotes": 1,
                "userId": 1,
              },
              Object {
                "addedBy": "admin367",
                "createdAt": "2018-02-20T10:57:03.524Z",
                "description": "awesome ",
                "downvotes": 0,
                "id": 48,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F99d42668-42c7-435a-9ea3-0d8ab7a0e62a.jpeg?alt=media&token=cc1d3c0a-716d-42ff-8222-693757baa5cb",
                "ingredients": "garri, vegetable,seasoning, blah,blah,blah",
                "mealType": "lunch",
                "method": "wash the vegatable
boil the water",
                "recipeName": "Eba and vegetable",
                "updatedAt": "2018-02-27T10:05:58.619Z",
                "upvotes": 2,
                "userId": 3,
              },
            ]
          }
        />
        <PopularRecipeCardList
          popularRecipes={
            Array [
              Object {
                "addedBy": "admin367",
                "createdAt": "2018-02-20T10:57:03.524Z",
                "description": "awesome ",
                "downvotes": 0,
                "id": 48,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F99d42668-42c7-435a-9ea3-0d8ab7a0e62a.jpeg?alt=media&token=cc1d3c0a-716d-42ff-8222-693757baa5cb",
                "ingredients": "garri, vegetable,seasoning, blah,blah,blah",
                "mealType": "lunch",
                "method": "wash the vegatable
boil the water",
                "recipeName": "Eba and vegetable",
                "updatedAt": "2018-02-27T10:05:58.619Z",
                "upvotes": 2,
                "userId": 3,
              },
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-02-20T10:10:45.197Z",
                "description": "amazing recipe",
                "downvotes": 1,
                "id": 26,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F702446f3-a2c2-48c7-a9d1-af20be9281d7.jpeg?alt=media&token=1827adb2-4108-4939-9fe0-778e27dd5212",
                "ingredients": "rice,chicken,rice,chicken",
                "mealType": "lunch",
                "method": "research on that",
                "recipeName": "rice and chicken",
                "updatedAt": "2018-03-12T09:32:31.056Z",
                "upvotes": 2,
                "userId": 1,
              },
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-02-27T09:54:05.790Z",
                "description": "An amazing recipe made from chicken",
                "downvotes": 0,
                "id": 53,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F401e40a2-a363-4910-9c94-2f5d2c5875b1.jpg?alt=media&token=acf1977d-c406-4a88-9577-9fa1e34c8f70",
                "ingredients": "chicken, vegetables, fruits",
                "mealType": "elevenses",
                "method": "slice the chicken to small pieces
wash the vegetables
slice the fruits into small pieces",
                "recipeName": "Shredded chiken",
                "updatedAt": "2018-03-06T10:56:57.198Z",
                "upvotes": 1,
                "userId": 1,
              },
              Object {
                "addedBy": "username",
                "createdAt": "2018-03-06T11:19:10.386Z",
                "description": "Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed. Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed. Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed. Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed.",
                "downvotes": 0,
                "id": 56,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F6f3fd226-4237-47fd-a716-f64b3c3dea2f.jpg?alt=media&token=f754b717-f91b-4bc7-9aeb-7fd19f98b123",
                "ingredients": "Some, elements, that, are, clickable, mysteriously, don't, trigger, a pointer cursor in browsers. This fixes that, and provides, a, default class, \\"pointer\\" for applying, it to other clickable ,things as needed",
                "mealType": "brunch",
                "method": "Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed. Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed. Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed. Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed.",
                "recipeName": "Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed.",
                "updatedAt": "2018-03-13T02:54:06.452Z",
                "upvotes": 1,
                "userId": 2,
              },
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-03-10T23:12:44.072Z",
                "description": "amazing",
                "downvotes": 0,
                "id": 59,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F0672f71e-5018-4a19-8970-ecc6c835d827.jpg?alt=media&token=a9730299-fad9-45ba-a07f-3ef5742f9092",
                "ingredients": "indomie",
                "mealType": "breakfast",
                "method": "google it",
                "recipeName": "indomie",
                "updatedAt": "2018-03-11T08:24:52.021Z",
                "upvotes": 1,
                "userId": 1,
              },
              Object {
                "addedBy": "admin367",
                "createdAt": "2018-02-20T18:17:37.078Z",
                "description": "cupcakes",
                "downvotes": 0,
                "id": 51,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2Ffb54ab3c-fd39-4955-a9a9-3d4282a6fc10.png?alt=media&token=3058df06-c8eb-461d-ad5b-197996221ad3",
                "ingredients": "cake,cake,cake,cake",
                "mealType": "tea",
                "method": "research on that",
                "recipeName": "cupcakes",
                "updatedAt": "2018-03-06T10:57:21.354Z",
                "upvotes": 1,
                "userId": 3,
              },
            ]
          }
        />
      </div>
    </div>
  </div>
  <div
    className="sticky-paginate"
  >
    <Pagination
      handlePaginationChange={[Function]}
      page={2}
    />
  </div>
  <Footer />
</div>
`;

exports[`RecipePage component should render correctly for user not logged in 1`] = `
<div>
  <Header />
  <div
    className="content"
  >
    <div
      className="container cont_area"
    >
      <div
        className="row"
      >
        <RecipeCardGrid
          allRecipes={
            Array [
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-02-20T10:10:45.197Z",
                "description": "amazing recipe",
                "downvotes": 1,
                "id": 26,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F702446f3-a2c2-48c7-a9d1-af20be9281d7.jpeg?alt=media&token=1827adb2-4108-4939-9fe0-778e27dd5212",
                "ingredients": "rice,chicken,rice,chicken",
                "mealType": "lunch",
                "method": "research on that",
                "recipeName": "rice and chicken",
                "updatedAt": "2018-03-12T09:32:31.056Z",
                "upvotes": 2,
                "userId": 1,
              },
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-02-20T10:11:25.020Z",
                "description": "amazing",
                "downvotes": 1,
                "id": 27,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F702446f3-a2c2-48c7-a9d1-af20be9281d7.jpeg?alt=media&token=1827adb2-4108-4939-9fe0-778e27dd5212",
                "ingredients": "awesome",
                "mealType": "breakfast",
                "method": "great",
                "recipeName": "new recipe",
                "updatedAt": "2018-03-06T11:08:34.739Z",
                "upvotes": 0,
                "userId": 1,
              },
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-02-20T10:19:50.363Z",
                "description": "test",
                "downvotes": 0,
                "id": 33,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F463fb7f5-37ad-4a76-8dce-2ad3feb73e00.jpeg?alt=media&token=6b3afedd-09f3-4582-829a-6a665454db8d",
                "ingredients": "great",
                "mealType": "tea",
                "method": "great",
                "recipeName": "recipe",
                "updatedAt": "2018-03-06T10:57:55.712Z",
                "upvotes": 1,
                "userId": 1,
              },
              Object {
                "addedBy": "admin367",
                "createdAt": "2018-02-20T10:57:03.524Z",
                "description": "awesome ",
                "downvotes": 0,
                "id": 48,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F99d42668-42c7-435a-9ea3-0d8ab7a0e62a.jpeg?alt=media&token=cc1d3c0a-716d-42ff-8222-693757baa5cb",
                "ingredients": "garri, vegetable,seasoning, blah,blah,blah",
                "mealType": "lunch",
                "method": "wash the vegatable
boil the water",
                "recipeName": "Eba and vegetable",
                "updatedAt": "2018-02-27T10:05:58.619Z",
                "upvotes": 2,
                "userId": 3,
              },
            ]
          }
        />
        <PopularRecipeCardList
          popularRecipes={
            Array [
              Object {
                "addedBy": "admin367",
                "createdAt": "2018-02-20T10:57:03.524Z",
                "description": "awesome ",
                "downvotes": 0,
                "id": 48,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F99d42668-42c7-435a-9ea3-0d8ab7a0e62a.jpeg?alt=media&token=cc1d3c0a-716d-42ff-8222-693757baa5cb",
                "ingredients": "garri, vegetable,seasoning, blah,blah,blah",
                "mealType": "lunch",
                "method": "wash the vegatable
boil the water",
                "recipeName": "Eba and vegetable",
                "updatedAt": "2018-02-27T10:05:58.619Z",
                "upvotes": 2,
                "userId": 3,
              },
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-02-20T10:10:45.197Z",
                "description": "amazing recipe",
                "downvotes": 1,
                "id": 26,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F702446f3-a2c2-48c7-a9d1-af20be9281d7.jpeg?alt=media&token=1827adb2-4108-4939-9fe0-778e27dd5212",
                "ingredients": "rice,chicken,rice,chicken",
                "mealType": "lunch",
                "method": "research on that",
                "recipeName": "rice and chicken",
                "updatedAt": "2018-03-12T09:32:31.056Z",
                "upvotes": 2,
                "userId": 1,
              },
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-02-27T09:54:05.790Z",
                "description": "An amazing recipe made from chicken",
                "downvotes": 0,
                "id": 53,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F401e40a2-a363-4910-9c94-2f5d2c5875b1.jpg?alt=media&token=acf1977d-c406-4a88-9577-9fa1e34c8f70",
                "ingredients": "chicken, vegetables, fruits",
                "mealType": "elevenses",
                "method": "slice the chicken to small pieces
wash the vegetables
slice the fruits into small pieces",
                "recipeName": "Shredded chiken",
                "updatedAt": "2018-03-06T10:56:57.198Z",
                "upvotes": 1,
                "userId": 1,
              },
              Object {
                "addedBy": "username",
                "createdAt": "2018-03-06T11:19:10.386Z",
                "description": "Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed. Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed. Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed. Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed.",
                "downvotes": 0,
                "id": 56,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F6f3fd226-4237-47fd-a716-f64b3c3dea2f.jpg?alt=media&token=f754b717-f91b-4bc7-9aeb-7fd19f98b123",
                "ingredients": "Some, elements, that, are, clickable, mysteriously, don't, trigger, a pointer cursor in browsers. This fixes that, and provides, a, default class, \\"pointer\\" for applying, it to other clickable ,things as needed",
                "mealType": "brunch",
                "method": "Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed. Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed. Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed. Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed.",
                "recipeName": "Some elements that are clickable mysteriously don't trigger a pointer cursor in browsers. This fixes that, and provides a default class \\"pointer\\" for applying it to other clickable things as needed.",
                "updatedAt": "2018-03-13T02:54:06.452Z",
                "upvotes": 1,
                "userId": 2,
              },
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-03-10T23:12:44.072Z",
                "description": "amazing",
                "downvotes": 0,
                "id": 59,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F0672f71e-5018-4a19-8970-ecc6c835d827.jpg?alt=media&token=a9730299-fad9-45ba-a07f-3ef5742f9092",
                "ingredients": "indomie",
                "mealType": "breakfast",
                "method": "google it",
                "recipeName": "indomie",
                "updatedAt": "2018-03-11T08:24:52.021Z",
                "upvotes": 1,
                "userId": 1,
              },
              Object {
                "addedBy": "admin367",
                "createdAt": "2018-02-20T18:17:37.078Z",
                "description": "cupcakes",
                "downvotes": 0,
                "id": 51,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2Ffb54ab3c-fd39-4955-a9a9-3d4282a6fc10.png?alt=media&token=3058df06-c8eb-461d-ad5b-197996221ad3",
                "ingredients": "cake,cake,cake,cake",
                "mealType": "tea",
                "method": "research on that",
                "recipeName": "cupcakes",
                "updatedAt": "2018-03-06T10:57:21.354Z",
                "upvotes": 1,
                "userId": 3,
              },
            ]
          }
        />
      </div>
    </div>
  </div>
  <div
    className="sticky-paginate"
  >
    <Pagination
      handlePaginationChange={[Function]}
      page={2}
    />
  </div>
  <Footer />
</div>
`;
